crates/holochain/src/conductor/handle.rs

```diff
@@ -633,10 +635,15 @@ impl<DS: DnaStore + 'static> ConductorHandleT for ConductorHandleImpl<DS> {
             } => {
                 use holochain_sqlite::db::AsP2pAgentStoreConExt;
                 let env = { self.p2p_env(space) };
-                let res = env
-                    .conn()?
-                    .p2p_gossip_query_agents(since_ms, until_ms, (*arc_set).clone())
-                    .map_err(holochain_p2p::HolochainP2pError::other);
+                let permit = env.conn_permit().await;
+                let res = tokio::task::spawn_blocking(move || {
+                    let mut conn = env.from_permit(permit)?;
+                    conn.p2p_gossip_query_agents(since_ms, until_ms, (*arc_set).clone())
+                })
+                .await;
```

Comment:
Is this not something that can be done inside `p2p_gossip_query_agents` so we don't forget to do it in other places?

Reply:
maybe different types for a conn that waits for a permit vs one that doesn't? and then only define `p2p_gossip_query_agents` on the one that does?

Reply:
That's what is happening. It's only implemented for the `PConnGuard`